src/Markup/Avalonia.Markup/Data/IndexerNode.cs

```diff
@@ -51,8 +52,110 @@ protected override IObservable<object> StartListeningCore(WeakReference referenc
             return Observable.Merge(inputs).StartWith(GetValue(target));
         }
 
+        public bool SetTargetValue(object value, BindingPriority priority)
+        {
+            var typeInfo = Target.Target.GetType().GetTypeInfo();
+            var list = Target.Target as IList;
+            var dictionary = Target.Target as IDictionary;
+            var indexerProperty = GetIndexer(typeInfo);
+            var indexerParameters = indexerProperty?.GetIndexParameters();
+
+            if (indexerProperty != null && indexerParameters.Length == Arguments.Count)
+            {
+                var convertedObjectArray = new object[indexerParameters.Length];
+
+                for (int i = 0; i < Arguments.Count; i++)
```

Comment:
Rather than doing this conversion every time a value is set, would it be best to store the converted arguments in `Arguments`? The arguments won't change, right?